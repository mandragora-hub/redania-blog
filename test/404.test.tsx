/* eslint-disable no-undef */

import { render, screen } from '@testing-library/react'

import NotFound from '../pages/404'

const useRouter = jest.spyOn(require('next/router'), 'useRouter')

describe('NotFound', () => {
  it('renders a 404 heading', () => {
    useRouter.mockImplementationOnce(() => ({
      query: { 404: 'not-found' },
    }))
    const { container } = render(<NotFound />)

    const heading = screen.getByRole('heading', {
      name: /404/i,
    })

    expect(heading).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })
})
