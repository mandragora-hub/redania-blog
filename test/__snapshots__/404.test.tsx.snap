// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound renders a 404 heading 1`] = `
<div>
  <div
    class="flex flex-col items-start justify-start md:mt-24 md:flex-row md:items-center md:justify-center md:space-x-6"
  >
    <div
      class="space-x-2 pt-6 pb-8 md:space-y-5"
    >
      <h1
        class="text-6xl font-extrabold leading-9 tracking-tight text-gray-900 dark:text-gray-100 md:border-r-2 md:px-6 md:text-8xl md:leading-14"
      >
        404
      </h1>
    </div>
    <div
      class="max-w-md"
    >
      <p
        class="mb-4 text-xl font-bold leading-normal md:text-2xl"
      >
        Sorry we couldn't find this page.
      </p>
      <p
        class="mb-8"
      >
        But don't worry, you can find plenty of other things on our homepage.
      </p>
      <a
        href="/"
      >
        <button
          class="focus:shadow-outline-blue inline rounded-lg border border-transparent bg-blue-600 px-4 py-2 text-sm font-medium leading-5 text-white shadow transition-colors duration-150 hover:bg-blue-700 focus:outline-none dark:hover:bg-blue-500"
        >
          Back to homepage
        </button>
      </a>
    </div>
  </div>
</div>
`;
